prime_server/http_protocol.hpp

```diff
@@ -32,7 +34,21 @@ namespace prime_server {
     std::string buffer;
   };
 
-  using headers_t = std::unordered_map<std::string, std::string>;
+  struct caseless_predicates_t : public std::hash<std::string> {
```

Comment:
lets us do caseless lookups in the header map, importantly it still stores the fields with whatever casing was supplied